compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -834,12 +837,22 @@ LogicalResult generateXCLBin(MLIRContext *context, ModuleOp moduleOp,
 LogicalResult AIETargetBackend::serializeExecutable(
     const SerializationOptions &serOptions,
     IREE::HAL::ExecutableVariantOp variantOp, OpBuilder &executableBuilder) {
+  // Create a flatbuffer containing xclbin and lx6 instructions, which gets
+  // insertered into the IR as an op attribute. The design here is copied from
+  // the CUDA plugin implementation, see
+  // iree/compiler/plugins/target/CUDA/CUDATarget.cpp
+  FlatbufferBuilder builder;
+  iree_amd_aie_hal_xrt_ExecutableDef_start_as_root(builder);
+
```

Comment:
I think it would read better if we do all the flatbuffer related activity at once in the end, so lets move this at the very end?